eos-updater/poll-common.c

```diff
@@ -383,9 +387,206 @@ get_booted_refspec (OstreeDeployment     *booted_deployment,
   return TRUE;
 }
 
+/* On split-disk systems, an additional (bigger, slower) disk is mounted at
+ * /var/endless-extra, and the system flatpak repo is configured to be at
+ * /var/endless-extra/flatpak rather than /var/lib/flatpak/repo. */
+static gboolean
+booted_system_is_split_disk (OstreeRepo *os_repo)
+{
+  g_autoptr(GUnixMountEntry) extra_mount = NULL, root_mount = NULL;
+
+  extra_mount = g_unix_mount_for ("/var/endless-extra", NULL);
+  root_mount = g_unix_mount_for ("/", NULL);
+
+  if (g_strcmp0 (g_getenv ("EOS_UPDATER_TEST_IS_SPLIT_DISK"), "1") == 0)
+    return TRUE;
+
+  return (g_unix_mount_compare (extra_mount, root_mount) != 0);
```

Comment:
I think this would be more straightforward to use `g_unix_mount_at ("/var/endless-extra", NULL)` and check if the result is `NULL` or not. I actually think this would produce false positives on non-split Endless systems, too. `/var` is a bind mount, so the entry returned by `g_unix_mount_for` will be `/var` in the non-split case, which will not match the mount at `/`. I'm pretty sure all systems will be regarded as split disk systems.

Reply:
> I don't understand why the `GUnixMountEntry` would be `NULL`. This probably has something to do with the chroot setup in jenkins.

I neglected to consider the case where either of the mount entries were `NULL`. It didn’t help that [GLib didn’t document this](https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2145) until an introspection annotation was added recently.

Your suggestion is good though, and avoids a load of faffing with handling different things possibly being `NULL`.